app/Filament/Pages/EditProfile.php

```diff
@@ -44,67 +49,103 @@ class EditProfile extends Page
 
     public function form(Form $form): Form
     {
+        /** @var User $user */
         $user = auth()->user();
 
+        $connectedAccounts = collect([
+            Grid::make()
+                ->schema([
+                    Placeholder::make('calendar')
+                        ->label(function (): string {
+                            /** @var User $user */
+                            $user = auth()->user();
+
+                            return str('Calendar')
+                                ->when($user->calendar->provider_type, fn (Stringable $str) => $str->prepend("{$user->calendar->provider_type->getLabel()} "))
+                                ->when($user->calendar->name, fn (Stringable $str) => $str->append(" - {$user->calendar->name}"))
+                                ->toString();
+                        }),
+                    Actions::make([
+                        FormAction::make('Disconnect')
+                            ->icon('heroicon-m-trash')
+                            ->color('danger')
+                            ->requiresConfirmation()
+                            ->action(function () {
+                                /** @var User $user */
+                                $user = auth()->user();
+
+                                $user->calendar->delete();
```

Comment:
We don't have to do anything about it now, we could create a backlog ticket for it. But is there anything on the OAuth API end where we can revoke the implementation?

That way the integration is deleted on the providers end as well?

Reply:
It appears that this is all I have to do so I just implemented it:
```php
public function revokeToken(Calendar $calendar): bool
{
    return static::client($calendar)
        ->revokeToken($calendar->oauth_token);
}
```

Reply:
I doubled checked it in: https://myaccount.google.com/connections and it does remove the app.